File: Objetivo 3/Objetivo3_Exc_L1_PHP/model/Gerente.php
<?php

include_once "Funcionario.php";

/**
 * Description of Gerente
 *
 * @author vagner
 */
class Gerente extends Funcionario{
    /*
     * Onde você deve inserir as regras de negócio? Sem dúvidas, é no modelo. O controlador só deve gerenciar o fluxo entre a Model e a View.
     */
    public function getBonus() {
        return parent::getSalario() * 0.20;
    }
    
    public function __toString() {
        return "Gerente{" . parent::__toString() . "}";
    }


}

File: Objetivo 3/Objetivo3_Exc_L1_Java/src/model/Gerente.java
package model;

/**
 *
 * @author vagner
 */
public class Gerente extends Funcionario{

    @Override
    public double getBonus() {
        return super.getSalario() * 0.20;
    }

    @Override
    public String toString() {
        return "Gerente{" + super.toString() + "}";
    }
    
}
